NullPointerException thrown converting Maven POM

<!--- 
Please follow the instructions below. 
We receive dozens of issues every week, so to stay productive, we will close issues that don't provide enough information. 

Please open Android-related issues on the Android Issue Tracker at https://source.android.com/source/report-bugs
Please open Gradle Native-related issues at https://github.com/gradle/gradle-native/issues
-->

<!--- Provide a brief summary of the issue in the title above -->

### Expected Behavior

Not to throw an exception when parsing my Maven `pom.xml`, and give clear error when something is wrong.

Simular to issue #15827

### Current Behavior

```
C:\Users\Borewit\IdeaProjects\lizzy> .\gradlew init

Found a Maven build. Generate a Gradle build from this? (default: yes) [yes, no]

Select build script DSL:                                                                                                                                                                          
  1: Groovy                                                                                                                                                                                       
  2: Kotlin
Enter selection (default: Groovy) [1..2] 1

Generate build using new APIs and behavior (some features may change in the next minor release)? (default: no) [yes, no]                                                                          
                                                                                                                                                                                                  
> Task :init
Maven to Gradle conversion is an incubating feature.
Get more help with your project: https://docs.gradle.org/8.0/userguide/migrating_from_maven.html

> Task :init FAILED

FAILURE: Build failed with an exception.

* What went wrong:

NullPointerException thrown running `gradle init`, converting from Maven.

Execution failed for task ':init'.
> A failure occurred while executing org.gradleinternal.buildinit.plugins.internal.maven.Maven2GradleWorkAction
   > Could not convert Maven POM C:\Users\Borewit\IdeaProjects\lizzy\pom.xml to a Gradle build.
      > Cannot invoke "String.trim()" because the return value of "org.codehaus.plexus.util.xml.Xpp3Dom.getValue()" is null

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s
2 actionable tasks: 1 executed, 1 up-to-date
```

Exception thrown here:
https://github.com/gradle/gradle/blob/270f942676177479a8e389cdd734a18771bc4982/subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/Maven2Gradle.java#L416


### Context
Trying to convert a Maven project to Gradle.

### Steps to Reproduce 

#### Summary

[pom.xml](https://github.com/Borewit/lizzy/blob/5138d8f4b6d2806d12b308f5f99801f70bb73678/pom.xml):
```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
<!-- ....... -->
  <build>
    <plugins>
      <plugin>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>3.1</version>
        <configuration>
          <source></source> <-- Empty source causes NullpointerException -->
          <target/>
        </configuration>
      </plugin>
    </plugins>
  </build>    
<!-- ....... -->
</project>
```
#### Detailed reproduction
1. Chechout https://github.com/Borewit/lizzy/commit/5138d8f4b6d2806d12b308f5f99801f70bb73678
2. Run gradle init

### Your Environment
* Windows
* Gradle 8.0

--- a/subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MavenConversionIntegrationTest.groovy
+++ b/subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MavenConversionIntegrationTest.groovy
@@ -694,6 +694,34 @@ Root project 'webinar-parent'
         dsl.getBuildFile().text.readLines().contains(descriptionPropertyAssignment)
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/23963")
+    def "emptySource"() {
+        def dsl = dslFixtureFor(scriptDsl)
+
+        when:
+        run 'init', '--dsl', scriptDsl.id as String
+
+        then:
+        dsl.assertGradleFilesGenerated()
+        dsl.getSettingsFile().text.contains("rootProject.name = 'util'") || dsl.getSettingsFile().text.contains('rootProject.name = "util"')
+        assertContainsPublishingConfig(dsl.getBuildFile(), scriptDsl)
+        succeeds 'clean', 'build'
+    }
+
+    @Issue("https://github.com/gradle/gradle/issues/23963")
+    def "emptyTarget"() {
+        def dsl = dslFixtureFor(scriptDsl)
+
+        when:
+        run 'init', '--dsl', scriptDsl.id as String
+
+        then:
+        dsl.assertGradleFilesGenerated()
+        dsl.getSettingsFile().text.contains("rootProject.name = 'util'") || dsl.getSettingsFile().text.contains('rootProject.name = "util"')
+        assertContainsPublishingConfig(dsl.getBuildFile(), scriptDsl)
+        succeeds 'clean', 'build'
+    }
+
     static libRequest(MavenHttpRepository repo, String group, String name, Object version) {
         MavenHttpModule module = repo.module(group, name, version as String)
         module.allowAll()

--- a/subprojects/build-init/src/integTest/resources/org/gradle/buildinit/plugins/MavenConversionIntegrationTest/emptySource/some-thing/pom.xml
+++ b/subprojects/build-init/src/integTest/resources/org/gradle/buildinit/plugins/MavenConversionIntegrationTest/emptySource/some-thing/pom.xml
@@ -1,0 +1,20 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>util</groupId>
+    <artifactId>util</artifactId>
+    <version>2.5</version>
+    <packaging>jar</packaging>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source/>
+                    <target>1.7</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/subprojects/build-init/src/integTest/resources/org/gradle/buildinit/plugins/MavenConversionIntegrationTest/emptyTarget/some-thing/pom.xml
+++ b/subprojects/build-init/src/integTest/resources/org/gradle/buildinit/plugins/MavenConversionIntegrationTest/emptyTarget/some-thing/pom.xml
@@ -1,0 +1,20 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>util</groupId>
+    <artifactId>util</artifactId>
+    <version>2.5</version>
+    <packaging>jar</packaging>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.7</source>
+                    <target/>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/Maven2Gradle.java
+++ b/subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/Maven2Gradle.java
@@ -413,10 +413,10 @@ public class Maven2Gradle {
             if (configuration != null) {
                 Xpp3Dom configuredSource = configuration.getChild("source");
                 Xpp3Dom configuredTarget = configuration.getChild("target");
-                if (configuredSource != null && !configuredSource.getValue().trim().isEmpty()) {
+                if (configuredSource != null && configuredSource.getValue() != null && !configuredSource.getValue().trim().isEmpty()) {
                     source = configuredSource.getValue();
                 }
-                if (configuredTarget != null && !configuredTarget.getValue().trim().isEmpty()) {
+                if (configuredTarget != null && configuredTarget.getValue() != null && !configuredTarget.getValue().trim().isEmpty()) {
                     target = configuredTarget.getValue();
                 }
             }

